atoma-proxy/src/server/http_server.rs

```diff
@@ -297,8 +309,91 @@ pub(crate) struct NodePublicAddressRegistrationOpenApi;
 #[instrument(level = "info", skip_all)]
 pub async fn node_public_address_registration(
     State(state): State<ProxyState>,
-    Json(payload): Json<NodePublicAddressAssignment>,
+    headers: HeaderMap,
+    request: Request<Body>,
 ) -> Result<Json<Value>, StatusCode> {
+    let base64_signature = headers
+        .get("X-Signature")
+        .ok_or_else(|| {
+            error!("Signature header not found");
+            StatusCode::BAD_REQUEST
+        })?
+        .to_str()
+        .map_err(|e| {
+            error!("Failed to extract base64 signature encoding, with error: {e}");
+            StatusCode::BAD_REQUEST
+        })?;
+
+    let body_bytes = axum::body::to_bytes(request.into_body(), MAX_BODY_SIZE)
+        .await
+        .map_err(|_| {
+            error!("Failed to convert body to bytes");
+            StatusCode::BAD_REQUEST
+        })?;
+
+    let signature = Signature::from_str(base64_signature).map_err(|_| {
+        error!("Failed to parse signature");
+        StatusCode::BAD_REQUEST
+    })?;
+
+    let public_key_bytes = signature.public_key_bytes();
+    let public_key =
+        SuiPublicKey::try_from_bytes(signature.scheme(), public_key_bytes).map_err(|e| {
+            error!("Failed to extract public key from bytes, with error: {e}");
+            StatusCode::BAD_REQUEST
+        })?;
+    let sui_address = SuiAddress::from(&public_key);
+
+    let mut blake2b_hash = Blake2b::new();
+    blake2b_hash.update(&body_bytes);
```

Comment:
what is the body look like ? can you give an example ?

Also should the node repository have a method to call this endpoint ?

Reply:
yes we should
The body is of type `NodePublicAddressAssignment`
```
pub struct NodePublicAddressAssignment {
    /// Unique small integer identifier for the node
    node_small_id: u64,
    /// The public address of the node
    public_address: String,
}
```